Update docs for Swift interceptors (#68)
connectrpc/connectrpc.com: docs/swift/interceptors.md

Interceptors are a powerful way to observe and mutate outbound and inbound
headers, data, trailers, and errors both for unary APIs and streams.

Each interceptor is instantiated **once per request or stream** and
provides a set of closures that are invoked by the client during the lifecycle
of that call. Each closure allows the interceptor to observe and store
state, as well as to mutate outbound or inbound content.

Every interceptor has the opportunity to perform asynchronous work before passing a potentially
altered value to the next interceptor in the chain. When the end of the chain is reached, the

> final value is passed to the networking client where it is sent to the server (outbound request)

Reviewer: ```suggestion
final value is passed to the networking client, where it is sent to the server (outbound request)
```

I think there should be a comma in this sentence before the 'where.'

or to the caller (inbound response).

> Interceptors may also fail outbound requests before they're sent, thus preventing subsequent
> interceptors from being invoked and returning a specified error back to the original caller.

Reviewer: ```suggestion
Interceptors may also fail outbound requests before they are sent; subsequent 
interceptors in the chain will not be invoked, and the error will be returned to the original caller.
```

I think removing the 'thus preventing' language better communicates that subsequent interceptors not being invoked is more just a side effect of the failure rather than the key purpose of the failure (which is, as stated, to prevent the *request* from being sent).

> Interceptors are closure-based and are passed both the current value and a closure which
> should be called to resume the interceptor chain. Propagation will not continue until
> this closure is called. Additional values may still be passed to a given interceptor even

Reviewer: ```suggestion
Interceptors are closure-based and receive both the current value and a closure that 
should be called to resume the interceptor chain. Propagation will not continue until 
this closure is invoked. Additional values may still be passed to a given interceptor even
```

though it has not yet continued the chain with a previous value. For example:

- A request is sent
- Response headers are received, and an interceptor pauses the chain while processing them
- First chunk of streamed data is received, and the interceptor is invoked with this value

> - Interceptor is expected to resume with headers first, and then with data after

Reviewer 2: Possibly better as an ordered list?

Author: As in - in the documentation, or resuming in code by passing a list?

Reviewer: I think @akshayjshah is referring to the docs here. Since those steps happen in sequence, it may make sense to represent them in an ordered list (1, 2, 3...).

Author: 👍🏽 updated
